trigger_take_weapon enhanced

Inky-Inky opened the pull request:
* Now able to take any weapon (not only the shotgun anymore)
* Fine control over the player's ammo inventory (add/subtract/reset)

Incidentally, I also fixed a ridiculous bug dating back to Quake 1.0: the maximum cells amount allowed is 100 and that value is enforced in many places, yet 200 units are given when cheating with `impulse 9`. So if you cheat then grab a shell box, your cells inventory suddenly drops back to 100! 🤦‍♂️

Inky-Inky committed:
trigger_take_weapon enhanced

Now able to take any weapon (not only the shotgun anymore) + control the player's ammo inventory

EmeraldTiger reviewed (changes requested):
Bug with taking away weapons: If the player is still carrying a weapon that gets removed, it will disappear from the inventory as intended, but the player will continue to wield it until they switch away from it.

Inky-Inky committed:
Fix weapon-only removal

The switch to another weapon occurred in case of an empty stock of ammo for the current weapon but was not triggered if the ammo was still there... but not the current weapon itself anymore!

EmeraldTiger reviewed (approved):
Fix confirmed
